GoogleForCreators/web-stories-wp
# Quick Actions: Foreground image actions

## Context

Setting up more of the quick actions

## Summary

Adds the quick actions when selecting a 'foreground image'.

## Relevant Technical Choices

n/a

## To-do


- [ ] hide snackbar when the user clicks undo. maybe show a snackbar stating that the animations were re-applied (doing in a separate pr)
- [x] finish karma tests for 'clear animations' action
- [x] update karma tests once #7522 is merged. Allows us to focus the tabs in the library when clicking a quick action.


## User-facing changes

<details>
<summary>Gifs are here but very large</summary>

|Action|Gif|
|--|--|
|Replace Media|![first-action](https://user-images.githubusercontent.com/22185279/118729372-08f2d500-b7f3-11eb-98c4-6e6fab54a18b.gif)|
|Add Animation|![second-action](https://user-images.githubusercontent.com/22185279/118729384-0ee8b600-b7f3-11eb-865b-768303e8ab9e.gif)|
|Add Link|![third-action](https://user-images.githubusercontent.com/22185279/118729389-127c3d00-b7f3-11eb-9e14-3453352fcc05.gif)|
|Clear Animations|![fourth-action](https://user-images.githubusercontent.com/22185279/118729393-14de9700-b7f3-11eb-84dd-dce46e16236a.gif)|
</details>

## Testing Instructions

1. Make sure quick actions experiment is enabled
2. Go to editor
3. Add image to the canvas (not as a background image)
4. Mess with actions - clicking and keyboard. Actions should do the following:
    - Replace media: should show the user the media3p panel in the library. Should focus the media3p tab.
    - Add animation: should focus the animation panel in the design panel. Should focus the dropdown.
    - Add link: should focus the link panel in the design panel. Should focus the input.
    - Clear animations: should remove the animations from the element. If there are no animations nothing should happen. Clicking `undo` in the snackbar that's shown should add the previously removed animation back to the element

### QA






### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6143
